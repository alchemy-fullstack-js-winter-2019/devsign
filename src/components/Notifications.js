import React from 'react';

export default function Notifications() {
  return (
    <article>
      <header>
        <h2>Notifications</h2>
      </header>
      <section>
        <ol>
          <li>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Hilly Jill</h3>
            <p>Liked your post</p>
          </li>
          <li>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Hilly Jill</h3>
            <p>New Tweet from Hilly Jill</p>
          </li>
        </ol>
      </section>
    </article>
  );
}
