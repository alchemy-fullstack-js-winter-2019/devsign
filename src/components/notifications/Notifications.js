import React from 'react';
import Notification from './Notification';
import PropTypes from 'prop-types';


const notes = [
  {
    user: { handle: 'Huey', image: 'https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg' },
    action: 'Liked your photo',
    date: 'March 12, 2019'
  },
  {
    user: { handle: 'Luey', image: 'https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg' },
    action: 'Tweeted',
    date: 'March 11, 2019'
  },
  {
    user: { handle: 'Duey', image: 'https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg' },
    action: 'Liked your tweet',
    date: 'March 12, 2019'
  }
];
export default function Notifications({ notifications = notes }) {
  const listOfNotifications = notifications.map((notification, i) => {
    return <li key={i}><Notification notification={notification} /></li>;
  });
  
  return (
    <article>
      <header>
        <h2>Notifications</h2>
      </header>
      <ul>{listOfNotifications}</ul>
    </article>
  );
}

Notifications.propTypes = {
  notifications: PropTypes.array
};
