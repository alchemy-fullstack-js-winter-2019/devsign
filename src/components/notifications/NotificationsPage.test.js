import React from 'react';
import renderer from 'react-test-renderer';
import Notifications from './Notifications';

describe('Notifications', () => {
  it('matches a snapshot', () => {
    const notifications = [];
    const tree = renderer.create(
      <Notifications notifications={notifications}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
