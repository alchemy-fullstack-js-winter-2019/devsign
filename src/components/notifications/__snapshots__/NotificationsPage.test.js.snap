// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications matches a snapshot 1`] = `
<article>
  <header>
    <h2>
      Notifications
    </h2>
  </header>
  <ul />
</article>
`;
